test/test_mcmc.py

```diff
@@ -11,28 +12,29 @@
 from numpyro.util import scan
 
 
-def test_unnormalized_normal():
+@pytest.mark.parametrize('algo', ['HMC', 'NUTS'])
+def test_unnormalized_normal(algo):
     true_mean, true_std = 1., 2.
     warmup_steps, num_samples = 1000, 8000
 
     def potential_fn(z):
-        return 0.5 * np.sum(((z[0] - true_mean) / true_std) ** 2)
+        return 0.5 * np.sum(((z - true_mean) / true_std) ** 2)
 
     def kinetic_fn(r, m_inv):
-        return 0.5 * np.sum(m_inv * r[0] ** 2)
+        return 0.5 * np.sum(m_inv * r ** 2)
 
-    init_kernel, sample_kernel = hmc_kernel(potential_fn, kinetic_fn)
-    init_samples = [np.array([0.])]
+    init_kernel, sample_kernel = hmc_kernel(potential_fn, kinetic_fn, algo)
+    init_samples = np.array(0.)
     hmc_state = init_kernel(init_samples,
+                            num_steps=10,
```

Comment:
Could you elaborate on why `num_steps` is needed? Is it for NUTS or HMC?

Reply:
I missed your comment above. Let me go through the gist.

Reply:
I added a comment above the instability of the default trajectory length